flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconciler.java

```diff
@@ -302,6 +304,14 @@ public boolean reconcileOtherChanges(FlinkResourceContext<FlinkDeployment> ctx)
         return cleanupTerminalJmAfterTtl(ctx.getFlinkService(), deployment, observeConfig);
     }
 
+    private String getMsgRestartUnhealthy(ClusterHealthInfo clusterHealthInfo) {
+        if (clusterHealthInfo != null) {
+            return MSG_RESTART_UNHEALTHY + " : " + clusterHealthInfo.getHealthResult().getError();
```

Comment:
The event reason already contains RESTARTUNHEALTHYJOB, we can probably simply have the error message here, and maybe get rid of this method completely?

Reply:
Thanks, this is a good shout, I have updated it